mod/redir.php

```diff
@@ -44,102 +47,105 @@ function redir_init(App $a) {
 	// Try magic auth before the legacy stuff
 	redir_magic($a, $cid, $url);
 
-	if (!empty($cid)) {
-		$fields = ['id', 'uid', 'nurl', 'url', 'addr', 'name', 'network', 'poll', 'issued-id', 'dfrn-id', 'duplex', 'pending'];
-		$contact = DBA::selectFirst('contact', $fields, ['id' => $cid, 'uid' => [0, local_user()]]);
-		if (!DBA::isResult($contact)) {
-			notice(DI::l10n()->t('Contact not found.'));
-			DI::baseUrl()->redirect();
-		}
+	if (empty($cid)) {
+		throw new \Friendica\Network\HTTPException\NotFoundException(DI::l10n()->t('Contact not found.'));
```

Comment:
This should be a `BadRequestException` instead.

Reply:
Okay.